lib/pf/util/dns.pm

```diff
@@ -56,6 +57,42 @@ sub matches_passthrough {
     return ($FALSE, []);
 }
 
+=head2 matches_isolation_passthrough
+
+Whether or not a domain matches a configured DNS isolation passthrough
+
+Returns 2 values
+- Whether or not it matched
+- List of ports that should be opened for this domain
+
+=cut
+
+sub matches_isolation_passthrough {
```

Comment:
We need to create a function that will match any passthroughs to a domain so that the code isn't a copy/paste
Let me know if you want me to do it or if you'll take care of it